jOOQ/src/main/java/org/jooq/impl/InsertImpl.java

```diff
@@ -580,6 +580,13 @@ public final InsertImpl defaultValues() {
 
     @Override
     public final InsertImpl doUpdate() {
+        onDuplicateKeyUpdate = true;
+        return this;
+    }
+
+    @Override
+    public final InsertImpl doIgnore() {
+        getDelegate().onDuplicateKeyIgnore(true);
         return this;
```

Comment:
I think that from a "principle of least surprise" point of view, we should now set `onDuplicateKeyUpdate` in both of these methods, and we should stop setting the flag in `onConflict()`